### PR TITLE
Feat: Fake Data in URL

### LINKED ISSUE #857: Use fakes in the handlebar template
As mentioned in [this discussion](https://github.com/hyperjumptech/monika/discussions/854), it'd be great if we can use some variables in header, url, and body. For example, 

```yaml
probes:
  - id: '1'
    name: 'Example: get Time'
    description: Probe
    interval: 10 # in seconds
    requests:
      - method: POST
        url: https://mybackend.org/user/login?time={{ timestamp() }} # replace timestamp with the actual timestamp when the request is sent
        timeout: 7000 # in milliseconds
        saveBody: true
        headers:
          Authorization: Bearer {{ alphaNumeric(30) }} # generate a random alpha numeric string
        body:
          username: {{ exampleEmail() }} # generate a random email address
          password: {{ {{ words(3) }} }} # generate 3 random words
        alerts:
          - query: response.status != 200
            message: Status not 2xx
    incidentThreshold: 3
    recoveryThreshold: 3
    alerts:
      - query: response.status != 200
        message: HTTP response status is {{ response.status }}, expecting 200
```

So, we should add the following features:

- [ ] Allow using `timestamp()` in the handlebar template. This will be replaced with the number of milliseconds elapsed since January 1, 1970 00:00:00 UTC.
- [ ] Allow using any functions from [fakerjs](https://fakerjs.dev/guide/). For example, `{{ uuid() }}` will call `faker.datatype.uuid()` from fakerjs.
- [ ] When the function is not recognized (not `timestamp()` nor any functions from fakerjs, Monika should show error.

### PR DESCRIPTION
**Pull request by dennypradipta**
# Monika Pull Request (PR)  

## What feature/issue does this PR add  
This PR closes #857 

## How did you implement / how did you fix it  
1. Add faker
2. Create fakes utils
3. Register the fakes utils as Handlebars helper

## How to test  
Try with this config:
```
notifications:
  - id: desktop
    type: desktop

probes:
  - id: '0'
    name: '0'
    interval: 10
    requests:
      - url: http://github.com?alpha={{ alpha }}
  - id: '1'
    name: '1'
    interval: 10
    requests:
      - url: http://github.com?alphaNumeric={{ alphaNumeric }}
  - id: '2'
    name: '2'
    interval: 10
    requests:
      - url: http://github.com?countryCode={{ countryCode }}
  - id: '3'
    name: '3'
    interval: 10
    requests:
      - url: http://github.com?color={{ color }}
  - id: '4'
    name: '4'
    interval: 10
    requests:
      - url: http://github.com?currency={{ currency }}
  - id: '5'
    name: '5'
    interval: 10
    requests:
      - url: http://github.com?email={{ email }}
  - id: '6'
    name: '6'
    interval: 10
    requests:
      - url: http://github.com?fullName={{ fullName }}
  - id: '7'
    name: '7'
    interval: 10
    requests:
      - url: http://github.com?gender={{ gender }}
  - id: '8'
    name: '8'
    interval: 10
    requests:
      - url: http://github.com?latitude={{ latitude }}
  - id: '9'
    name: '9'
    interval: 10
    requests:
      - url: http://github.com?lines={{ lines }}
  - id: '10'
    name: '10'
    interval: 10
    requests:
      - url: http://github.com?longitude={{ longitude }}
  - id: '11'
    name: '11'
    interval: 10
    requests:
      - url: http://github.com?number={{ number }}
  - id: '12'
    name: '12'
    interval: 10
    requests:
      - url: http://github.com?objectId={{ objectId }}
  - id: '13'
    name: '13'
    interval: 10
    requests:
      - url: http://github.com?statusCode={{ statusCode }}
  - id: '14'
    name: '14'
    interval: 10
    requests:
      - url: http://github.com?timestamp={{ timestamp }}
  - id: '15'
    name: '15'
    interval: 10
    requests:
      - url: http://github.com?uuid={{ uuid }}
  - id: '16'
    name: '16'
    interval: 10
    requests:
      - url: http://github.com?word={{ word }}
  - id: '17'
    name: '17'
    interval: 10
    requests:
      - url: http://github.com?words={{ words }}
```

image

Supported fake types:
```
alpha:
alphaNumeric:
countryCode:
color:
currency:
email:
fullName:
gender:
latitude:
lines:
longitude:
number:
objectId:
statusCode:
timestamp:
uuid:
word:
words:
```

For the moment, the props passed to the function are hardcoded. Still figuring out on how to pass the props

### CODE REVIEW

**Review by nicnocquee — CHANGES_REQUESTED** [review]

File: src/components/probe/probing.ts [hunk 1]
```diff
@@ -61,6 +66,8 @@ export async function probing({
   const renderURL = Handlebars.compile(url)
   const renderedURL = renderURL({ responses })
 
+  console.log(renderedURL)
```
[the remark this answers is not in the document]
Comment (dennypradipta) [reply src/components/probe/probing.ts]: That was a mistake for debugging, my bad. Should be fixed

**Comment by dennypradipta** [comment]
Added more fake data:
```
  countryCode,
  color,
  currency,
  email,
  gender,
  latitude,
  longitude,
  number,
  objectId,
  statusCode,
  timestamp,
  uuid,
```

**Comment by dennypradipta** [comment]
Updated. Try with this config:
notifications:
  - id: desktop
    type: desktop

probes:
  - id: '1'
    requests:
      - url: http://github.com?alpha={{ alpha }}
      - url: http://github.com?alphaNumeric={{ alphaNumeric }}
      - url: http://github.com?countryCode={{ countryCode }}
      - url: http://github.com?color={{ color }}
      - url: http://github.com?currency={{ currency }}
      - url: http://github.com?email={{ email }}
      - url: http://github.com?fullName={{ fullName }}
      - url: http://github.com?gender={{ gender }}
      - url: http://github.com?latitude={{ latitude }}
      - url: http://github.com?longitude={{ longitude }}
      - url: http://github.com?number={{ number }}
      - url: http://github.com?objectId={{ objectId }}
      - url: http://github.com?statusCode={{ statusCode }}
      - url: http://github.com?timestamp={{ timestamp }}
      - url: http://github.com?uuid={{ uuid }}
      - url: http://github.com?word={{ word }}
      - url: http://github.com?words={{ words }}

Supported fake types:
```
alpha:
alphaNumeric:
countryCode:
color:
currency:
email:
fullName:
gender:
latitude:
lines:
longitude:
number:
objectId:
statusCode:
timestamp:
uuid:
word:
words:
```

For the moment, the props passed to the function are hardcoded. Still figuring out on how to pass the props

**Comment by dennypradipta** [comment]
Updated. Try with this config:
```
notifications:
  - id: desktop
    type: desktop

probes:
  - id: '0'
    name: '0'
    interval: 10
    requests:
      - url: http://github.com?alpha={{ alpha }}
  - id: '1'
    name: '1'
    interval: 10
    requests:
      - url: http://github.com?alphaNumeric={{ alphaNumeric }}
  - id: '2'
    name: '2'
    interval: 10
    requests:
      - url: http://github.com?countryCode={{ countryCode }}
  - id: '3'
    name: '3'
    interval: 10
    requests:
      - url: http://github.com?color={{ color }}
  - id: '4'
    name: '4'
    interval: 10
    requests:
      - url: http://github.com?currency={{ currency }}
  - id: '5'
    name: '5'
    interval: 10
    requests:
      - url: http://github.com?email={{ email }}
  - id: '6'
    name: '6'
    interval: 10
    requests:
      - url: http://github.com?fullName={{ fullName }}
  - id: '7'
    name: '7'
    interval: 10
    requests:
      - url: http://github.com?gender={{ gender }}
  - id: '8'
    name: '8'
    interval: 10
    requests:
      - url: http://github.com?latitude={{ latitude }}
  - id: '9'
    name: '9'
    interval: 10
    requests:
      - url: http://github.com?lines={{ lines }}
  - id: '10'
    name: '10'
    interval: 10
    requests:
      - url: http://github.com?longitude={{ longitude }}
  - id: '11'
    name: '11'
    interval: 10
    requests:
      - url: http://github.com?number={{ number }}
  - id: '12'
    name: '12'
    interval: 10
    requests:
      - url: http://github.com?objectId={{ objectId }}
  - id: '13'
    name: '13'
    interval: 10
    requests:
      - url: http://github.com?statusCode={{ statusCode }}
  - id: '14'
    name: '14'
    interval: 10
    requests:
      - url: http://github.com?timestamp={{ timestamp }}
  - id: '15'
    name: '15'
    interval: 10
    requests:
      - url: http://github.com?uuid={{ uuid }}
  - id: '16'
    name: '16'
    interval: 10
    requests:
      - url: http://github.com?word={{ word }}
  - id: '17'
    name: '17'
    interval: 10
    requests:
      - url: http://github.com?words={{ words }}
```

image

Supported fake types:
```
alpha:
alphaNumeric:
countryCode:
color:
currency:
email:
fullName:
gender:
latitude:
lines:
longitude:
number:
objectId:
statusCode:
timestamp:
uuid:
word:
words:
```

For the moment, the props passed to the function are hardcoded. Still figuring out on how to pass the props

**Comment by dennypradipta** [comment]
Also to be noted, because it is changed in the `probing.ts` file, I think it could also apply to the request body or headers. Kindly check it

**Comment by dennypradipta** [comment]
Updated. Added `isostring`, dynamic params, and improve docs.

Use this newer config:
```
notifications:
  - id: desktop
    type: desktop

probes:
  - id: '0'
    name: '0'
    interval: 20
    requests:
      - url: http://github.com?alpha={{ alpha }}
      - url: http://github.com?alpha={{ alpha 8 }}
  - id: '1'
    name: '1'
    interval: 20
    requests:
      - url: http://github.com?alphaNumeric={{ alphaNumeric }}
      - url: http://github.com?alphaNumeric={{ alphaNumeric 8 }}
  - id: '2'
    name: '2'
    interval: 10
    requests:
      - url: http://github.com?countryCode={{ countryCode }}
  - id: '3'
    name: '3'
    interval: 10
    requests:
      - url: http://github.com?color={{ color }}
  - id: '4'
    name: '4'
    interval: 10
    requests:
      - url: http://github.com?currency={{ currency }}
  - id: '5'
    name: '5'
    interval: 10
    requests:
      - url: http://github.com?email={{ email }}
  - id: '6'
    name: '6'
    interval: 10
    requests:
      - url: http://github.com?fullName={{ fullName }}
  - id: '7'
    name: '7'
    interval: 10
    requests:
      - url: http://github.com?gender={{ gender }}
  - id: '7.5'
    name: '5'
    interval: 10
    requests:
      - url: http://github.com?isostring={{ isostring }}
  - id: '8'
    name: '8'
    interval: 30
    requests:
      - url: http://github.com?latitude={{ latitude }}
      - url: http://github.com?latitude={{ latitude 40 }}
      - url: http://github.com?latitude={{ latitude 80 80 }}
  - id: '9'
    name: '9'
    interval: 20
    requests:
      - url: http://github.com?lines={{ lines }}
      - url: http://github.com?lines={{ lines 3 }}
  - id: '10'
    name: '10'
    interval: 30
    requests:
      - url: http://github.com?longitude={{ longitude }}
      - url: http://github.com?latitude={{ longitude 40 }}
      - url: http://github.com?latitude={{ longitude 80 80 }}
  - id: '11'
    name: '11'
    interval: 30
    requests:
      - url: http://github.com?number={{ number }}
      - url: http://github.com?number={{ number 8 }}
      - url: http://github.com?number={{ number 8 36 }}
  - id: '12'
    name: '12'
    interval: 10
    requests:
      - url: http://github.com?objectId={{ objectId }}
  - id: '13'
    name: '13'
    interval: 10
    requests:
      - url: http://github.com?statusCode={{ statusCode }}
  - id: '14'
    name: '14'
    interval: 10
    requests:
      - url: http://github.com?timestamp={{ timestamp }}
  - id: '15'
    name: '15'
    interval: 10
    requests:
      - url: http://github.com?uuid={{ uuid }}
  - id: '16'
    name: '16'
    interval: 10
    requests:
      - url: http://github.com?word={{ word }}
  - id: '17'
    name: '17'
    interval: 20
    requests:
      - url: http://github.com?words={{ words }}
      - url: http://github.com?words={{ words 3 }}
```

image

File: docs/src/pages/guides/probes.md [hunk 2]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+- `{{ fullName }}` returns a full name
+
+  e.g `{{ fullName }}` returns `John Doe`
+
+- `{{ gender }}` returns a gender
+
+  e.g `{{ gender }}` returns `Male/Female`
+
+- `{{ isostring }}` returns a date in ISO String format
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:191]: this is not random date right? but the date of the request? if so, `returns the date of the request sending in ISO format`

File: docs/src/pages/guides/probes.md [hunk 3]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+- `{{ latitude <min> <max> }}` returns a latitude between `min` and `max` value
+
+  The default value for `min` and `max` are `-90` and `90`.
+
+  e.g `{{ latitude }}` returns `-30.9501`, `{{ latitude 0 30 }}` returns `3.9521`
+
+- `{{ lines <lineCount> }}` returns `<lineCount>` lines of lorem ipsum lines
+
+  e.g `{{ lines }}` returns `Lorem ipsum dolor sit amet`, `{{ lines 2 }}` returns `Commodi non ex voluptatibus quibusdam nisi aliquam dolor nihil. Eos maiores enim praesentium.`
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:203]: return `lineCount` number of lines of "Lorem ipsum" strings

File: docs/src/pages/guides/probes.md [hunk 4]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+- `{{ longitude <min> <max> }}` returns a longitude between `min` and `max` value
+
+  The default value for `min` and `max` are `-180` and `180`.
+
+  e.g `{{ longitude }}` returns `-30.9501`, `{{ longitude 0 30 }}` returns `3.9521`
+
+- `{{ number <min> <max> }}` returns a random number range from `min` to `max`
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:211]: returns a random integer number between min and max

File: docs/src/pages/guides/probes.md [hunk 5]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+  The default value for `min` and `max` are `0` and `1000`.
+
+  e.g `{{ number }}` returns `720`, `{{ number 0 500 }}` returns `480`
+
+- `{{ objectId }}` returns a MongoDB ObjectID
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:217]: returns a random MongoDB ObjectID

File: docs/src/pages/guides/probes.md [hunk 6]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+  e.g `{{ objectId }}` returns `63034695eca3670c4e083657`
+
+- `{{ statusCode }}` returns a status code
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:221]: returns a random HTTP status code

File: docs/src/pages/guides/probes.md [hunk 7]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+  e.g `{{ statusCode }}` returns `200`
+
+- `{{ timestamp }}` returns current time in a form of a UNIX timestamp
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:225]: returns the time of the request sending in UNIX timestamp format

File: docs/src/pages/guides/probes.md [hunk 8]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+  e.g `{{ timestamp }}` returns `1661159139803`
+
+- `{{ uuid }}` returns a UUID
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:229]: returns a random UUID

File: docs/src/pages/guides/probes.md [hunk 9]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+  e.g `{{ uuid }}` returns `d8ffd51c-88cd-453e-906e-389b145891e7`
+
+- `{{ word }}` returns a word
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:233]: returns a random English word

File: docs/src/pages/guides/probes.md [hunk 10]
```diff
@@ -148,6 +148,151 @@ Currently, Monika only supports Content-Type value `application/x-www-form-urlen
...
+
+  e.g `{{ word }}` returns `nicely`
+
+- `{{ words <count> }}` returns `<count>` words
```
Comment (nicnocquee) [inline docs/src/pages/guides/probes.md:237]: returns `count` number of random English words

**Review by nicnocquee — CHANGES_REQUESTED** [review]

**Comment by dennypradipta** [comment]
Updated docs
image

Along with test
image

**Review by nicnocquee — APPROVED** [review]

**Review by sapiderman — APPROVED** [review]

**Pull request merged**